assignments/HW4.md

```diff
@@ -0,0 +1,194 @@
+
+You should be documenting your API endpoints with [Swagger](https://swagger.io/docs/). A helpful resource to achieve this in with a Flask App can be found [here](https://www.aurigait.com/blog/api-documentation-with-swagger-in-flask/). The API themselves don't need to be doing anything at the moment. Just make sure you define the request/response body fields and headers, if necessary.
+
+Because this is just a checkpoint, the expectation is that you will hit this checkpoint, and then continue to work. Therefore, when you have completed this checkpoint, you should tag a version of your code as a release for version 0.1. You can find instructions as to how to do this [here](https://help.github.com/en/articles/creating-releases).
+
+Then, submit this tagged version and a pdf of your Swagger documentation that describes the API endpoints to the associated Gradescope assignment.
+You can create a pdf of your Swagger documentation with the Print Preview function of your browser.
+
+### (2) Microservice ML feature implementation - (due Tuesday, Nov 1st 11:59 PM) | 55 points
+
+We provide you with a starter repository (see above) with a (synthetic) historical dataset (as a csv file) of student performance (this is an augmented version of the dataset found here), and a (bad) model trained over it. Start by exploring the data, to understand what information is available. Student performance is described by their grades (G1, G2, and G3), where G3 is the final year grade, the target performance metric. Grades are on a scale of 0-20; 20 is the best. CMU considers a high-quality student one with a G3 grade of 15 or higher.
+
+The existing model in the repository is very simple. Once you have familiarized yourself with the dataset, explore this model using whatever means you think are appropriate and determine in which ways you don’t believe it to be suitable. Based on what you learn in this exploratory step, you should train a new model and build a deployable microservice.
+
+You should evaluate your new model, and should explain how this model (overall) performs better than the baseline model, and any ways in which your model performs worse than the baseline model.
+
+Plan some testing, perform that testing on your microservice endpoints, and document it. At least some of this testing should be automated using Github Actions.
+As you have in previous assignments, you can create a test suite (ie. [pytest](https://flask.palletsprojects.com/en/2.2.x/testing/)) and have your workflow run the test suite.
+
+The deliverable for this task is a documented technical artifact; submit a link of your github repository to the Gradescope assignment. This should include:
+
+1. a trained model that is deployable as a microservice; the microservice must take as input a query request with student data and return a prediction about that student’s graduate school success. This is the API for your microservice. It should also include
```

Comment:
If we're providing a trained model, we might want to adjust the wording here to avoid any potential confusion

Reply:
The default model isn't trained in the starter code. We actually do prompt the students to cross validate (or to come up with a different model)